osquery/management.py

```diff
@@ -21,6 +25,90 @@
 from osquery.extension_client import ExtensionClient, DEFAULT_SOCKET_PATH
 from osquery.extension_manager import ExtensionManager
 
+DARWIN_BINARY_PATH = "/usr/local/bin/osqueryd"
+LINUX_BINARY_PATH = "/usr/bin/osqueryd"
+
+class SpawnInstance(object):
+    """Spawn a standalone osquery instance"""
+
+    '''The osquery process instance.'''
+    instance = None
+    '''The extension client connection attached to the instance.'''
+    connection = None
+    _socket = None
+
+    def __init__(self, path=None):
+        """
+        Keyword arguments:
+        path -- the path to and osqueryd binary to spawn
+        """
+        if path is None:
+            # Darwin is special and must have binaries installed in /usr/local.
+            if sys.platform == "darwin":
+                self.path = DARWIN_BINARY_PATH
+            else:
+                self.path = LINUX_BINARY_PATH
+        else:
+            self.path = path
+        self._socket = tempfile.mkstemp(prefix="pyosqsock")
+        self._pidfile = tempfile.mkstemp(prefix="pyosqpid")
+        with open(self._pidfile[1], "w") as fh:
+            fh.write("100000")
+        self._dbpath = tempfile.mkdtemp(prefix="pyoqsdb")
+
+    def __del__(self):
+        if self.connection is not None:
+            self.connection.close()
+        if self.instance is not None:
+            self.instance.kill()
+        # Todo: remove socket, pidfile, and database
+
+    def open(self, timeout=2, interval=0.01):
```

Comment:
Yeah, exception all the things. If you want to be really fancy, you can make custom exceptions so that people can handle the two cases separately, but I don't think that's really necessary.